idlc_codegen/src/marking.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum MarkingStyle {
+    Rust,
+    C,
+    Java,
+}
+
+impl MarkingStyle {
+    const fn start(self) -> &'static str {
+        match self {
+            Self::Rust | Self::C => "",
+            Self::Java => "/*\n",
+        }
+    }
+
+    const fn end(self) -> &'static str {
+        match self {
+            Self::Rust | Self::C => "",
+            Self::Java => "*/",
+        }
+    }
+
+    const fn prefix(self) -> &'static str {
+        match self {
+            Self::Rust | Self::C => "//",
+            Self::Java => "*",
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Marking(String);
+impl std::fmt::Display for Marking {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Display::fmt(&self.0, f)
+    }
+}
+
+impl std::ops::Deref for Marking {
+    type Target = str;
+
+    fn deref(&self) -> &Self::Target {
+        self.0.as_str()
+    }
+}
+impl AsRef<str> for Marking {
+    fn as_ref(&self) -> &str {
+        self.0.as_str()
+    }
+}
+
+impl Marking {
+    pub fn add_marking(marking: Option<std::path::PathBuf>, style: MarkingStyle) -> Self {

```

Comment:
I would call this `new(marking: &str, style: MarkingStyle) -> Self` and let the `main.rs` in `idlc` handle the write out, that way we can test this API easily. This function essentially shouldn't be called if we don't have a marking.